base_decorator.rb

```diff
@@ -0,0 +1,10 @@
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
```

Comment:
- [ ] Please require nameable here also because  BaseDecorator is making use of it. If the file is not required, you will encounter an `uninitialized constant Nameable` error when testing.

```ruby
require './nameable.rb'

class BaseDecorator < Nameable
   # content goes here
end
```